src/main/scala/ai/privado/languageEngine/java/threatEngine/CustomPrivacyLoggerMustbeUsed.scala

```diff
@@ -0,0 +1,55 @@
+package ai.privado.languageEngine.java.threatEngine
+
+import ai.privado.exporter.ExporterUtility
+import ai.privado.languageEngine.java.threatEngine.ThreatUtility.{hasDataElements}
+import ai.privado.model.{Constants, PolicyOrThreat}
+import ai.privado.model.exporter.ViolationProcessingModel
+import io.shiftleft.codepropertygraph.generated.Cpg
+import io.shiftleft.semanticcpg.language._
+import org.slf4j.LoggerFactory
+
+import scala.collection.mutable.ListBuffer
+import scala.util.Try
+
+object CustomPrivacyLoggerMustbeUsed {
+
+  private val logger        = LoggerFactory.getLogger(getClass)
+  private val MAX_INSTANCES = 5
+
+  /** Check for violation for data leakage to logs threat - consumes already generated dataflows
+    *
+    * @param cpg
+    *   cpg
+    * @return
+    */
+  def getViolations(threat: PolicyOrThreat, cpg: Cpg): Try[(Boolean, List[ViolationProcessingModel])] = Try {
+    if (hasDataElements(cpg)) {
+      val higherOrderLeakgeSinkId = "Leakages.Log.*"
+      val violatingFlows          = ListBuffer[ViolationProcessingModel]()
+      val CUSTOM_LOOGER_PATTERN   = getCustomLoggerModulePattern(threat.config)
+      val leakageSinks            = cpg.call.where(_.tag.nameExact(Constants.id).value(higherOrderLeakgeSinkId)).l
+
+      leakageSinks.foreach((leakage) => {
+        if (!leakage.methodFullName.matches(CUSTOM_LOOGER_PATTERN)) {
+          if (violatingFlows.size < MAX_INSTANCES) {
+            violatingFlows.append(
+              ViolationProcessingModel(leakage.name, ExporterUtility.convertIndividualPathElement(leakage).get)
+            )
+          }
+        }
+      })
+
+      (violatingFlows.nonEmpty, violatingFlows.toList)
+    } else (false, List())
+  }
+
+  private def getCustomLoggerModulePattern(config: Map[String, String]): String = {
+    val DEFAULT_PATTERN              = ""
+    val CUSTOM_LOGGER_MODULE_PATTERN = "customLoggerModulePattern"
+    if (config.contains(CUSTOM_LOGGER_MODULE_PATTERN)) {
+      config.get(CUSTOM_LOGGER_MODULE_PATTERN).getOrElse(DEFAULT_PATTERN)
```

Comment:
what regex you expect here for `CUSTOM_LOGGER_MODULE_PATTERN`?

Reply:
it will be generic pattern anything can come to match methodFullName from Threat rule.
In Infosys case its:
(?i)(portfolio).*(?:error|severe|fatal|warn|debug|trace|info|log|exception)